compiler/testData/codegen/box/jvm8/kt29242.kt

```diff
@@ -0,0 +1,24 @@
+// TARGET_BACKEND: JVM
+// IGNORE_BACKEND: JVM
+// JVM_TARGET: 1.8
```

Comment:
`JVM_TARGET ` is redundant. We have special test suite that set target to 1.8 in codegenTarget8Jvm8Test. And run configuration for it in https://teamcity.jetbrains.com/buildConfiguration/Kotlin_KotlinPublic_CodegenTestsOnDifferentJDKs?branch=

`TARGET_BACKEND` could be also removed

Reply:
So it's best to convert these tests to boxInline

Reply:
Are you sure this is true for local runs? Without JVM_TARGET, these tests pass on the non-IR backend. (I'm running them as `./gradlew cleanTest :compiler:test --parallel`). The rest of the files in this directory seem to have both JVM_TARGET and TARGET_BACKEND as well.

Reply:
Moved to boxInline,

Reply:
All of these tests are require //JVM_TARGET: 1.8 cause they wouldn't work on default target 1.6. And as result it's implies JVM target restriction.

And yes, you are right. This test would be passed on default target and failed on custom one. 
OK, let's keep tests as is. And after fixing problem in current backend I will convert this tests into boxInline.

Reply:
Moved back to box/jvm8.